Mathlib/CategoryTheory/WithTerminal/Cones.lean

```diff
@@ -0,0 +1,154 @@
+    | of a, star, _ => by aesop
+    | star, of _, _ => by contradiction
+    | of a, of b , f => by
+      have := by
+        calc
+          (t.π.app b).left = (t.π.app a ≫ K.map f).left := by
+            simp only [Functor.const_obj_obj, Cone.w]
+          _ = (t.π.app a).left ≫ (K.map f).left := rfl
+      simpa
+  }
+  map {t₁ t₂} f := {
+    hom := f.hom.left
+    w
+    | star => by aesop_cat
+    | of a => by
+      have := by calc
+        f.hom.left ≫ (t₂.π.app a).left = (f.hom ≫ t₂.π.app a).left := by rfl_cat
+        _ = (t₁.π.app a).left := by simp_all only [ConeMorphism.w, Functor.const_obj_obj]
+      simpa
+  }
+
+/-- This is the inverse of the previous construction: a cone of an extended functor
+`liftFromOver.obj K : WithTerminal J ⥤ C` consists of an object of `C`, together
+with morphisms. This same object is a cone of the original functor `K : J ⥤ Over X`. -/
+@[simps map obj_π]
+def coneBack {X : C} {K : J ⥤ Over X} : Cone (liftFromOver.obj K) ⥤ Cone K where
+  obj t := {
+    pt := Over.mk (t.π.app star)
+    π.app a := {
+      left := t.π.app (of a)
+      right := 𝟙 _
+      w := by
+        have := by
+          calc
+            t.π.app (of a) ≫ (K.obj a).hom = t.π.app (of a) ≫
+              (liftFromOver.obj K).map (homFrom a) := rfl
+            _ = t.π.app star := by simp only [Functor.const_obj_obj, Cone.w]
+        simpa
+    }
+    π.naturality a b f := by
+      ext
+      let f₂ := incl.map f
+      have eq_after_K: (K.map f₂).left = (K.map f).left := by aesop
+      have nat : t.π.app (of b) =
+        t.π.app (of a) ≫ (K.map f₂).left := by simpa using t.π.naturality f₂
+      simp [nat, eq_after_K]
+  }
+  map {t₁ t₂ f} := {
+    hom := Over.homMk f.hom
+  }
+
+@[simp]
+theorem coneBack_obj_pt {X : C} {K : J ⥤ Over X} (t : Cone (liftFromOver.obj K)) :
+    (coneBack.obj t).pt  = Over.mk (t.π.app star) := rfl
+
+/-- The isomorphism between `coneLift ⋙ coneBack` and the identity, at the level of objects. -/
+@[simps]
+def coneLiftBack {X : C} {K : J ⥤ Over X} (t : Cone K) : coneBack.obj (coneLift.obj t) ≅ t where
+  hom.hom := 𝟙 t.pt
+  inv.hom := 𝟙 t.pt
+
+/-- The isomorphism between `coneBack ⋙ coneLift` and the identity, at the level of objects. -/
+@[simps]
+def coneBackLift {X : C} {K : J ⥤ Over X} (t : Cone (liftFromOver.obj K)) :
+coneLift.obj (coneBack.obj t) ≅ t where
+  hom.hom := 𝟙 t.pt
+  inv.hom := 𝟙 t.pt
+
+/-- The equivalence made up of `coneBack` and `coneLift`. -/
+def coneEquiv {X : C} (K : J ⥤ Over X) : Cone K ≌ Cone (liftFromOver.obj K) where
+  functor := coneLift
+  inverse := coneBack
+  unitIso := NatIso.ofComponents coneLiftBack
+  counitIso := NatIso.ofComponents coneBackLift
+
+/-- A cone `t` of `K : J ⥤ Over X` is a limit if and only if the corresponding cone
+`coneLift t` of `liftFromOver.obj K : WithTerminal K ⥤ C` is a limit. -/
+def limitEquiv {X : C} {K : J ⥤ Over X} {t : Cone K} :
+  IsLimit (coneLift.obj t) ≃ IsLimit t := IsLimit.ofConeEquiv (coneEquiv K)
```

Comment:
Do we need this? It's a straightforward application of `IsLimit.ofConeEquiv`

Reply:
Do you have any examples? I hardly ever saw such defs.

Reply:
Sorry, am out of computer, but search for `ofConeEquiv` and you'll find a few